### PR TITLE
[NFC] test cleanup. Uses CRM_Core_Exceptions, properly outputs unfiltered results

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor cleanup in tests class

Before
----------------------------------------
More variants of Exception in play, printing unfiltered results on fail doesn't work

After
----------------------------------------
Above fixed

Technical Details
----------------------------------------
The comment blocks should show all exceptions thrown by a function so trying to stdise a little.

Comments
----------------------------------------

### CODE REVIEW

File: Civi/Test/Api3TestTrait.php [hunk 1]
```diff
@@ -214,8 +218,8 @@ public function callAPISuccessGetSingle($entity, $params, $checkAgainst = NULL)
     ];
     $result = $this->civicrm_api($entity, 'getsingle', $params);
     if (!is_array($result) || !empty($result['is_error']) || isset($result['values'])) {
-      $unfilteredResult = $this->civicrm_api($entity, 'get', $params);
-      throw new \Exception(
+      $unfilteredResult = $this->civicrm_api($entity, 'get', ['version' => $this->_apiversion]);
```
Comment (seamuslee001) [inline Civi/Test/Api3TestTrait.php]: @eileenmcnaughton this seems to be problematic in that there maybe additional params passed in that we are no not passing through to the API call

Comment (eileenmcnaughton) [reply Civi/Test/Api3TestTrait.php]: @seamuslee001 right - the point of this call was always to dump out what is in the DB - unfiltered - for that entity - but it wasn't working due to the fact it was being filtered

File: Civi/Test/ContactTestTrait.php [hunk 2]
```diff
@@ -151,15 +152,15 @@ public function sampleContact($contact_type, $seq = 0, $random = FALSE) {
...
+   * @throws CRM_Core_Exception
    *
    * @return int
    *   id of Household created
    */
   private function _contactCreate($params) {
     $result = $this->callAPISuccess('contact', 'create', $params);
     if (!empty($result['is_error']) || empty($result['id'])) {
-      throw new \Exception('Could not create test contact, with message: ' . \CRM_Utils_Array::value('error_message', $result) . "\nBacktrace:" . \CRM_Utils_Array::value('trace', $result));
+      throw new CRM_Core_Exception('Could not create test contact, with message: ' . CRM_Utils_Array::value('error_message', $result) . "\nBacktrace:" . CRM_Utils_Array::value('trace', $result));
```
Comment (seamuslee001) [inline Civi/Test/ContactTestTrait.php]: This needs a \ added in front of both function calls as we are in the Civi/Test/ name space

**Comment by eileenmcnaughton** [comment]
@seamuslee001 a bit delayed but I think I've fixed now

**Comment by seamuslee001** [comment]
Changes look good to me merging

**Pull request merged**